app/models/manageiq/providers/redhat/infra_manager/inventory/strategies/v4.rb

```diff
@@ -87,6 +87,8 @@ def collect_vms
     def collect_vm_by_uuid(uuid)
       vm = connection.system_service.vms_service.vm_service(uuid).get
       [VmPreloadedAttributesDecorator.new(vm, connection)]
+    rescue OvirtSDK4::Error
+      []
```

Comment:
Actually this is currently the right way to check if a VM exists using the oVirt Ruby SDK. The `vm_service(uuid)` method will always succeed, even if the VM doesn't exist. Calling the `get` method will then raise an exception, corresponding to the `Not found` HTTP code. The details of the error are in the exception message which isn't easy to parse.

I'd suggest to keep this code as it is. We will improve the SDK to provide a mechanism to check the kind of error, then this can be improved. For example:

```ruby
begin
  ...
rescue OvirtSDK4::Error => e
  if e.code == 404
    []
  else
    raise
  end
end
```
I have opened the following oVirt SDK bug to track it:

[RFE] Add a mechanism to check the HTTP result code of errors
https://bugzilla.redhat.com/1443420